iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -159,20 +159,39 @@ public override async Task SendEventAsync(IEnumerable<Message> messages, Cancell
             }
         }
 
-        public override async Task<Message> ReceiveAsync(TimeSpan timeout, CancellationToken cancellationToken)
+        public override async Task<Message> ReceiveAsync(CancellationToken cancellationToken)
         {
-            if (Logging.IsEnabled) Logging.Enter(this, timeout, cancellationToken, $"{nameof(ReceiveAsync)}");
+            if (Logging.IsEnabled) Logging.Enter(this, cancellationToken, $"{nameof(ReceiveAsync)}");
+
+            var defaultReceiveTimeout = TimeSpan.FromSeconds(10);
+
             Message message = null;
+
             while (true)
             {
                 cancellationToken.ThrowIfCancellationRequested();
-                message = await _amqpUnit.ReceiveMessageAsync(timeout).ConfigureAwait(false);
+                // The AMQP protocol implementation does not take a cancellation token. 
+                // We use a shorter timeout to be able to check on the cancellation state of the passed
+                // in token
+                message = await _amqpUnit.ReceiveMessageAsync(defaultReceiveTimeout).ConfigureAwait(false);
```

Comment:
Shorter times is the whole point of this PR. I need sub second response time when there are no messages in the queue.

Reply:
The cost with shorter time is to very expensive and it's going to impact some major products. As a trade off, if there is tight restrict on time, please use timeout instead of CancellationToken

Reply:
https://github.com/Azure/azure-iot-sdk-csharp/pull/1027 says this:
> Deprecating ReceiveAsync(Timeout): we recommend using the CancellationToken APIs instead.
Maybe I'm misinterpreting the meaning of that statement, but doesn't it mean that we shouldn't use timeout?

Reply:
I disagree to deprecate timeout API since AMQP library doesn't support `CancellationToken`.